draft-ietf-quic-recovery.md

```diff
@@ -460,21 +463,28 @@ The PTO period is the amount of time that a sender ought to wait for an
 acknowledgement of a sent packet.  This time period includes the estimated
 network roundtrip-time (smoothed_rtt), the variance in the estimate (4*rttvar),
 and max_ack_delay, to account for the maximum time by which a receiver might
-delay sending an acknowledgement.
+delay sending an acknowledgement.  When the PTO is armed for Initial or
+Handshake packet number spaces, the max_ack_delay is 0.
 
 The PTO value MUST be set to at least kGranularity, to avoid the timer expiring
 immediately.
 
+A sender computes its PTO timer every time an ack-eliciting packet is sent.
+When ack-eliciting packets are in-flight in multiple packet number spaces,
+the timer MUST be set for the packet number space with the earliest timeout,
+except for ApplicationData, which MUST be ignored until the handshake
+completes (See Section 4.1.1 of {{QUIC-TLS}}).  Not arming the PTO for
+ApplicationData prioritizes completing the handshake and prevents the server
+from sending a 1-RTT PTO packet before it has the keys to process a 1-RTT
+packet.
+
 When a PTO timer expires, the PTO period MUST be set to twice its current
 value. This exponential reduction in the sender's rate is important because
 the PTOs might be caused by loss of packets or acknowledgements due to severe
-congestion.  The life of a connection that is experiencing consecutive PTOs is
-limited by the endpoint's idle timeout.
-
-A sender computes its PTO timer every time an ack-eliciting packet is sent. A
-sender might choose to optimize this by setting the timer fewer times if it
-knows that more ack-eliciting packets will be sent within a short period of
-time.
+congestion.  Even when there are ack-eliciting packets in-flight in multiple
+packet number spaces, the exponential increase in probe timeout occurs across
+all spaces to prevent excess load on the network. The life of a connection that
```

Comment:
"For example, a timeout in the Initial packet number space doubles the length of the timeout in the Handshake number space."

Reply:
There is also some conceptual subtlety that I suspect people could mess up.

If I'm coalescing my probes, then there is *never* a timeout at the higher levels -- each time, the junior level times out and a new high-level packet is sent that resets that level.

Misinterpreting this paragraph might mean that coalescing three probes could result in three bitshifts of the PTO.

This doesn't have to be in this PR, but when I reread this draft when everything's landed, hopefully it's in here somewhere.

Reply:
Hopefully it's clearer now.